src/lib.rs

```diff
@@ -187,9 +186,9 @@ impl<T: Send> ThreadLocal<T> {
     where
         F: FnOnce() -> T,
     {
-        unsafe {
-            self.get_or_try(|| Ok::<T, ()>(create()))
-                .unchecked_unwrap_ok()
```

Comment:
I think you can even eliminate the `Err` line entirely.

Reply:
I get "missing match arm" on the current compiler version. I think you need [`exhaustive_patterns`](https://github.com/rust-lang/rust/issues/51085) to land first before that works.